### ISSUE TITLE
 Geo.searchByText return error "Cannot read properties of undefined"

### ISSUE THREAD

**Issue by Brendy89tm** [aws-amplify/amplify-js#12870]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

Angular

### Amplify APIs

Authentication, Geo

### Amplify Version

v6

### Amplify Categories

geo

### Backend

None

### Environment information

<details>

```
# Put output below this line
 System:
    OS: Windows 11 10.0.22621
    CPU: (8) x64 Intel(R) Core(TM) i5-1035G1 CPU @ 1.00GHz
    Memory: 1.10 GB / 7.75 GB
  Binaries:
    Node: 20.10.0 - C:\Program Files\nodejs\node.EXE
    npm: 8.18.0 - ~\AppData\Roaming\npm\npm.CMD
  Browsers:
    Edge: Chromium (120.0.2210.144)
    Internet Explorer: 11.0.22621.1
  npmPackages:
    @angular-devkit/build-angular: ^17.0.9 => 17.0.10
    @angular/animations: ^17.0.0 => 17.0.9
    @angular/cli: ^17.0.9 => 17.0.10
    @angular/common: ^17.0.0 => 17.0.9
    @angular/compiler: ^17.0.0 => 17.0.9
    @angular/compiler-cli: ^17.0.0 => 17.0.9
    @angular/core: ^17.0.0 => 17.0.9
    @angular/forms: ^17.0.0 => 17.0.9
    @angular/platform-browser: ^17.0.0 => 17.0.9
    @angular/platform-browser-dynamic: ^17.0.0 => 17.0.9
    @angular/router: ^17.0.0 => 17.0.9
    @maplibre/maplibre-gl-geocoder: ^1.5.0 => 1.5.0
    @maplibre/maplibre-gl-style-spec:  14.1.0-dev
    @types/jasmine: ~5.1.0 => 5.1.4
    @types/mapbox__mapbox-gl-draw: ^1.4.6 => 1.4.6
    aws-amplify: ^6.0.12 => 6.0.12
    aws-amplify/adapter-core:  undefined ()
    aws-amplify/analytics:  undefined ()
    aws-amplify/analytics/kinesis:  undefined ()
    aws-amplify/analytics/kinesis-firehose:  undefined ()
    aws-amplify/analytics/personalize:  undefined ()
    aws-amplify/analytics/pinpoint:  undefined ()
    aws-amplify/api:  undefined ()
    aws-amplify/api/server:  undefined ()
    aws-amplify/auth:  undefined ()
    aws-amplify/auth/cognito:  undefined ()
    aws-amplify/auth/cognito/server:  undefined ()
    aws-amplify/auth/enable-oauth-listener:  undefined ()
    aws-amplify/auth/server:  undefined ()
    aws-amplify/datastore:  undefined ()
    aws-amplify/in-app-messaging:  undefined ()
    aws-amplify/in-app-messaging/pinpoint:  undefined ()
    aws-amplify/push-notifications:  undefined ()
    aws-amplify/push-notifications/pinpoint:  undefined ()
    aws-amplify/storage:  undefined ()
    aws-amplify/storage/s3:  undefined ()
    aws-amplify/storage/s3/server:  undefined ()
    aws-amplify/storage/server:  undefined ()
    aws-amplify/utils:  undefined ()
    bootstrap: ^5.3.2 => 5.3.2
    jasmine-core: ~5.1.0 => 5.1.1 (4.6.0)
    karma: ~6.4.0 => 6.4.2
    karma-chrome-launcher: ~3.2.0 => 3.2.0
    karma-coverage: ~2.2.0 => 2.2.1
    karma-coverage-coffee-example:  1.0.0
    karma-jasmine: ~5.1.0 => 5.1.0
    karma-jasmine-html-reporter: ~2.1.0 => 2.1.0
    maplibre-gl: ^1.15.3 => 1.15.3
    maplibre-gl-js-amplify: ^4.0.0 => 4.0.0
    rxjs: ~7.8.0 => 7.8.1
    rxjs/ajax:  undefined ()
    rxjs/fetch:  undefined ()
    rxjs/operators:  undefined ()
    rxjs/testing:  undefined ()
    rxjs/webSocket:  undefined ()
    tslib: ^2.3.0 => 2.6.2 (1.14.1)
    typescript: ~5.2.2 => 5.2.2
    zone.js: ~0.14.2 => 0.14.3
  npmGlobalPackages:
    @angular/cli: 17.0.9
    npm-check-updates: 16.8.0
    npm: 8.18.0
    typescript: 4.8.2

```

</details>

### Describe the bug

On dummy project to understand how to use and implement AWS Location and its libraries, I have configured the Amplify with Amplify.configure( _amplyConfig_ ) but when try to search location with Geo.searchByText(text, options?) I give an error: TypeError: Cannot read properties of undefined (reading 'default')

### Expected behavior

the response of Geo.searchByText(text, options?) should give an array of results

### Reproduction steps

1) I have configured the Amplify with Amplify.configure( _amplyConfig_ )
2) _amplyConfig_  is set like this:
  ```js
  amplyConfig: ResourcesConfig = {
    Auth: {
      Cognito: {
        identityPoolId: '_pool_ID_',
        allowGuestAccess: true,
      },
    },
    Geo: {
      LocationService: {
        region: 'eu-central-1',
        maps: {
          items: {
            MappaTest: {
            },
          },
          default: 'MappaTest',
        },
        searchIndices: {
          items: ['MyMappaIndex'],
          default: 'MyMappaIndex',
        },
      },
    },
  };
```
3) note that, **amplifyConfig** is **ResourcesConfig** type from type.d.ts file
4) Geo field, based on _ResourcesConfig_ type is type _GeoConfig_
5) _GeoConfig_ is: 
```js
export type LocationServiceConfig = {
    LocationService: {
        region: string;
        maps?: {
            items: {};
            default: string;
        };
        searchIndices?: {
            items: string[];
            default: string;
        };
        geofenceCollections?: {
            items: string[];
            default: string;
        };
    };
};
export type GeoConfig = AtLeastOne<LocationServiceConfig>;
```
in detail **searchIndices** is **CamelCase**

6) when try to use  Geo.searchByText(text, options?) I give an error:
```
TypeError: Cannot read properties of undefined (reading 'default')
    at _AmazonLocationServiceProvider.<anonymous> (AmazonLocationServiceProvider.mjs:76:52)
    at Generator.next (<anonymous>)
    at fulfilled (chunk-3KMTWR7B.js?v=33b7c186:72:24)
    at _ZoneDelegate.invoke (zone.js:368:26)
    at Object.onInvoke (core.mjs:14714:33)
    at _ZoneDelegate.invoke (zone.js:367:52)
    at _Zone.run (zone.js:129:43)
    at zone.js:1257:36
    at _ZoneDelegate.invokeTask (zone.js:402:31)
    at core.mjs:14388:55
```
7) AmazonLocationServiceProvider.mjs:76:52 in detail here it try to seach **search_indices** SnakeCase
```js
let locationServiceInput = {
            Text: text,
            IndexName: this._config.search_indices.default
}
```
![image](https://github.com/aws-amplify/amplify-js/assets/157357185/940c3d57-ed04-4b04-91ab-b3ac7c48dc48)

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

**Comment by nadetastic** [comment]
Hi @Brendy89tm thank you for opening this issue, I have taken a look at this and I'm able to reproduce the issue.

The problem is that manually adding Geo to `Amplify.configure()` is causing the config not to be passed through in an expected way, but works when the config is set from the Amplify CLI generated `amplifyconfiguration.json` file.

During the invocation of the `Geo.searchByText()` api, the library expects the default index to be accessible from `config.search_indices.default`, however when passing the config manually, it appears in `config.searchIndices.default`, which causes the library to through the undefined error message.

I've marked this as a bug and will discuss with the team and follow up soon.

In the meantime, a potential work around is to mimic the shape of `amplifyconfiguration.json` and pass that in instead which may be able to get you unblocked. For example based on the config you shared above, this can translate to the following:

```js

const myconfigJson = {
  "aws_project_region": "__region__",
  "aws_cognito_identity_pool_id": "__pool-ID__",
  "aws_cognito_region": "__region__",
  "geo": {
    "amazon_location_service": {
      "region": "eu-central-1",
      "maps": {
        "items": {
          "MappaTest": {
          }
        },
        "default": "MappaTest"
      },
      "search_indices": {
        "items": [
          "MappaTest"
        ],
        "default": "MappaTest"
      }
    }
  }
}


// Pass the above config object into Amplify.configure()
Amplify.configure(configJson)
```

Please note that the above is only a temporary work around to get you going and I'll be working with the team to fix the bug.

**Comment by Brendy89tm** [comment]
thank you for your quickly reply! yes I am able to bypass the issue like your example or by not adding the type on my object.

**Comment by cwomack** [comment]
@Brendy89tm, the fix related to this issue was included in PR #13303 and should be resolved as of v6.2.0.
